src/term.c

```diff
@@ -964,7 +964,16 @@ static int cmd_erase(const PROGRAMMER *pgm, const AVRPART *p, int argc, const ch
     return cmd_write(pgm, p, 6, args);
   }
 
-  term_out("erasing chip ...\n");
+  const char *note = "";
+  if(avr_locate_eeprom(p)) {
+    int eesave, bakverb = verbose;
+    verbose = -123;             // Silently read EESAVE configuration bit
+    if(avr_get_config_value(pgm, p, "eesave", &eesave) == 0)
```

Comment:
What happens for (ancient) devices that don't have this configuration value?

Reply:
Here are those parts *with* EEPROM but *without* (known) `eesave` configuration

```
for p in $(avrdude -p*/St | grep eeprom.size | ct -f2); do
  echo $p $(avrdude -c dryrun -qqp $p -T "con eesave" 2>&1)
done |& grep -P "(error|warning)"
```

They fall into two categories:
 - Ancient device that truly doesn't have the configuration: `ATtiny12`, `ATtiny15`, `AT90S1200`, `AT90S4414`, `AT90S2313`, `AT90S2333`, `AT90S2343`, `AT90S2323`, `ATtiny22`, `AT90S4433`, `AT90S8515`, `AT90S8535`, `AT90S4434`, `ATmega163`, `ATmega161`
 - No ATDF file and/or no data sheet for neither love nor money (Part never existed? some knockoff part with emulated EEPROM?): `ATxmega64A4`, `ATxmega128A4`, `ATxmega192A1`, `ATxmega256A1`, `AVR8EA28`, `AVR8EA32`, `LGT8F88P`, `LGT8F168P`, `LGT8F328P`

Either way, the PR will say erasing chip ... without asserting anything about EEPROM, ie, neither claims erasing chip (including EEPROM) ... nor claims erasing chip (but not EEPROM) ...

@dl8dtl But what *do* these parts to on a CE? Erase EEPROM?

Reply:
$ avrdude -qqp t22 -c dryrun -T erase
erasing chip ...

$avrdude -qqp t26 -c dryrun -T erase
erasing chip (including EEPROM) ...
```

Reply:
... but I just realised that `erase` shouldn't affirm whether or not EEPROM is erased if the programmer is a bootloader; what the bootloader does with EEPROM on a CE is anyone's guess. So fixed *that*.

Reply:
> But what _do_ these parts do on a CE? Erase EEPROM?

Sure. That's the secure default that prevents leaking information if you had locked the device.

Reply:
I'm surprised about some of the missing ATDF files.

Reply:
> what the bootloader does with EEPROM on a CE is anyone's guess

It's not even sure the bootloader does anything at all when instructed to erase "the chip".